src/db.rs

```diff
@@ -0,0 +1,315 @@
+use openssl::encrypt::{Decrypter, Encrypter};
+use openssl::pkey::{PKey, Private};
+use openssl::rsa::Padding;
+use std::include_str;
+use tokio_postgres::{Error, Row, Transaction};
+
+const _CREATE_DB: &str = include_str!("queries/create_db.sql");
+const _DROP_DB: &str = include_str!("queries/drop_db.sql");
+const _CREATE_TABLE_CHAT: &str = include_str!("queries/create_table_chat.sql");
+const _CREATE_TABLE_CITIES: &str = include_str!("queries/create_table_cities.sql");
+const DELETE_CLIENT: &str = include_str!("queries/delete_client.sql");
+const GET_CITY_BY_PATTERN: &str = include_str!("queries/get_city_by_pattern.sql");
+const INSERT_CLIENT: &str = include_str!("queries/insert_client.sql");
+const IS_IN_DB: &str = include_str!("queries/is_in_db.sql");
+const MODIFY_CITY: &str = include_str!("queries/modify_city.sql");
+const MODIFY_CONTEXT: &str = include_str!("queries/modify_context.sql");
+const MODIFY_PATTERN_SEARCH: &str = include_str!("queries/modify_pattern_search.sql");
+const MODIFY_SELECTED: &str = include_str!("queries/modify_selected.sql");
+const MODIFY_STATE: &str = include_str!("queries/modify_state.sql");
+const SEARCH_CITY: &str = include_str!("queries/search_city.sql");
+const SEARCH_CLIENT: &str = include_str!("queries/search_client.sql");
+
+async fn _migrate(db_transaction: &mut Transaction<'_>) -> Result<(), Error> {
+    db_transaction.execute(_CREATE_TABLE_CHAT, &[]).await?;
+    db_transaction.execute(_CREATE_TABLE_CITIES, &[]).await?;
+    Ok(())
+}
+async fn _setup_db(mut db_transaction: Transaction<'_>) -> Result<(), Error> {
+    db_transaction.execute(_CREATE_DB, &[]).await?;
+    _migrate(&mut db_transaction).await?;
+    db_transaction.commit().await
+}
+async fn _rollback(db_transaction: Transaction<'_>) -> Result<(), Error> {
+    db_transaction.rollback().await
+}
+async fn _drop_db(db_transaction: Transaction<'_>) -> Result<(), Error> {
+    db_transaction.execute(_DROP_DB, &[]).await?;
+    db_transaction.commit().await
+}
+
+// Encrypt a String into a BYTEA
+async fn encrypt_string(some_string: String, keypair: &PKey<Private>) -> Vec<u8> {
```

Comment:
what do you think about instead of using functions, we can use some kind of struct and add methods into its impl

the struct can contain a pool of connections - the same way it's implemented in fang

Reply:
Solve in [Struct in Db and Pool , bb8_postgres crate](https://github.com/pxp9/weather_bot_rust/pull/6/commits/06f32062c8137184d170de857411a88189cce26c)